Tools/clinic/clinic.py

```diff
@@ -4427,13 +4521,19 @@ class DSLParser:
     state: StateKeeper
     keyword_only: bool
     positional_only: bool
+    deprecated_positional: VersionTuple | None
     group: int
     parameter_state: ParamState
     indent: IndentStack
     kind: FunctionKind
     coexist: bool
     parameter_continuation: str
     preserve_output: bool
+    deprecate_posonly_re = create_regex(
```

Comment:
posonly?

Reply:
Perhaps `star_from_version_re` is better?

Reply:
Suggesting `depr_star_re`.